src/lib/functions/netlify-function.mjs

```diff
@@ -93,11 +103,22 @@ export default class NetlifyFunction {
       const srcFilesDiff = this.getSrcFilesDiff(srcFilesSet)
 
       this.buildData = buildData
+      this.buildError = null
       this.srcFiles = srcFilesSet
       this.schedule = schedule || this.schedule
 
+      if (!this.isSupported()) {
+        throw new Error(
+          `Function requires Node.js version ${V2_MIN_NODE_VERSION} or above, but ${nodeVersion.slice(
+            1,
+          )} is installed. Refer to https://ntl.fyi/functions-node18 for information on how to update.`,
```

Comment:
Let's open an issue to make this link clickable. This would probably need changes here: https://github.com/netlify/cli/blob/9629588af0c30e4c07ed5c249c0d1ba5da5bb27c/src/lib/templates/function-error.html#L279-L280